### PR TITLE
Feature/add calico role

### PR DESCRIPTION
**Pull request by loshkovskyi**
Calico is a new virtual network solution, and it's added in the project
to enable the IP per container functionality.

etcd is added in the project as a key-value storage and a service
discovery mechanism for Calico.

### CODE REVIEW

**Comment by siddharthist** [comment]
Basic tests (Consul health checks, UIs, etc.) look good on:
 * Openstack

**Comment by siddharthist** [comment]
Failing health check on Vagrant
```
$ curl -k -L localhost:8500/v1/health/state/critical
[{"Node":"default","CheckID":"service:etcd:1","Name":"Service 'etcd' check","Status":"critical","Notes":"","Output":"unknown\n","ServiceID":"etcd","ServiceName":"etcd"},{"Node":"default","CheckID":"service:etcd:2","Name":"Service 'etcd' check","Status":"critical","Notes":"","Output":"/bin/sh: /usr/local/bin/consul-check-etcd-member: No such file or directory\n","ServiceID":"etcd","ServiceName":"etcd"}]
```

**Comment by loshkovskyi** [comment]
Thank you @siddharthist , addressed by https://github.com/CiscoCloud/microservices-infrastructure/commit/7223706d76ff536c0cb53eae33bb75053afb384c
Tested on Vagrant, OpenStack and DigitalOcean.

**Comment by siddharthist** [comment]
Tested and working on:
 * Vagrant

**Comment by siddharthist** [comment]
Just a note: if/when this is merged, we should open a ticket against distributive-rpm to include checklists for Calico and etcd. It'd be great if you could help with that @loshkovskyi. Gotta keep those roles healthy, too! :hospital:

**Comment by siddharthist** [comment]
On GCE, Marathon doesn't seem to be deploying marathon-consul and mesos-consul properly. 
screenshot

**Comment by loshkovskyi** [comment]
I have just deployed this branch on GCE. Both services: marathon-consul and mesos-consul are running just fine. I'm not able to reproduce this problem, sorry.

**Comment by siddharthist** [comment]
@loshkovskyi I'll try to reproduce again today.

**Comment by siddharthist** [comment]
@loshkovskyi I can definitely reproduce the issue. Are there any logs, etc. you'd like me to post from a deployment?

**Comment by loshkovskyi** [comment]
@siddharthist , yes, it would be helpful. You deploy on VMs which serve for testing deployments only, right? Maybe, you could provide me with temporary access to those VMs and I could investigate this issue much more faster. Otherwise, I'd like to check the logs from Mesos, Marathon, etcd, Calico and Docker. Thank you.

**Comment by loshkovskyi** [comment]
@siddharthist , regarding Distributive: added such task to my tasks list. Will apreciate your help on this.

**Comment by siddharthist** [comment]
@loshkovskyi Please post your preferred SSH public key, and I'll give you access once I build another.

**Comment by stevendborrelli** [comment]
Did we get issues resolved for this? I was wondering when it could be merged.

**Comment by siddharthist** [comment]
No, we have not gotten feedback on the apparent issue with Marathon tasks.

**Comment by stevendborrelli** [comment]
@loshkovskyi, @siddharthist  is the original Calico deployment even supposed to run on GCE?

**Comment by loshkovskyi** [comment]
@stevendborrelli, yes it should run on GCE just fine. In fact, I've never run into any issues with Calico on GCE. That's why I asked to provide me with the logs or grant access to your testing environment to troubleshoot the issue. Anyway, I'll do a couple of more testing deployments tomorrow.

**Comment by ryane** [comment]
should it also run on aws? I am actually seeing the same issue that @siddharthist described on an aws cluster.

**Comment by ryane** [comment]
I actually ran into the same problem on openstack too.

As hinted at in the readme, this depends on mesos 0.23.0. When running 0.22.1 (which we are pinned to in #618), you will see marathon-consul and mesos-consul stuck during deployment in marathon. With 0.23.0, I successfully tested this PR on aws, gce, and openstack, including the new calico functionality. In order to merge this, we are going to have to upgrade mesos to 0.23.0. So far, everything appears to be working fine on 0.23.0 but I'll do a little more testing tomorrow.

File: roles/calico/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+calico_image: ciscocloud/calico-node
+calico_image_tag: v0.4.9-mi
+
+bintray_baseurl: "https://bintray.com/artifact/download/loshkovskyi/generic"
```
Comment (ryane) [inline roles/calico/defaults/main.yml]: should we use the ciscocloud bintray account here?

Comment (loshkovskyi) [inline roles/calico/defaults/main.yml]: Still waiting for access to upload the files. Will ask if there are any updates.

**Comment by loshkovskyi** [comment]
@ryane, I will sync with the latest master and open a new PR.

**Comment by loshkovskyi** [comment]
New PR https://github.com/CiscoCloud/microservices-infrastructure/pull/658

**Pull request closed without merging**